### PR TITLE
Change ._data to ._parent for accessors

### PR DESCRIPTION
**Pull request by jbrockmendel**
The idea is to reduce the number of distinct meanings `._data` has.  With this it is down to just `Index._data` and `NDFrame._data`, I think.

### CODE REVIEW

**Comment by jreback** [comment]
I can see changing this as ``_parent`` is more clear; is it intention to change other uses of ``_data``?

**Comment by jbrockmendel** [comment]
> is it intention to change other uses of _data?

Not really.  At the sprint I brought up the idea of changing BlockManager references from _data to something more distinctive like _mgr, but the reception was luke-warm at best.

There is also #19658 but AFAIK there has been no action on that.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/accessors.py [hunk 1]
```diff
@@ -27,14 +27,14 @@ def __init__(self, data, orig):
             raise TypeError("cannot convert an object of type {0} to a "
                             "datetimelike index".format(type(data)))
 
-        self.values = data
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/core/indexes/accessors.py]: >  i think that the dateitmelike accessors do something different?

CategoricalAccessor uses `self.categories`, is the outlier this PR leaves unchanged.

**Comment by jbrockmendel** [comment]
The CI is having unrelated errors here much more consistently than usual.

**Comment by jbrockmendel** [comment]
@mroeschke if you're comfortable with the "Take easy stuff off jreback's plate" strategy, this may qualify.

**Comment by jreback** [comment]
i don’t think we should do this though

let me look a bit later

**Comment by jbrockmendel** [comment]
> i don’t think we should do this though

Huh, [here](https://github.com/pandas-dev/pandas/pull/21906#issuecomment-405028499) you seemed on board.  My mistake.

> let me look a bit later

The goal was to take "easy" stuff off your plate.  This is definitely not a priority.

**Comment by jreback** [comment]
@jbrockmendel hah, right I do remember that now. let me look again. my worry is that we are changing *some* things to ``._parent``, which makes sense, but now we have a dicotomy and now unclear what means what. maybe survey where we use ._data and propose a logical split?

**Comment by TomAugspurger** [comment]
I think using `._parent` for accessors (and *only* for accessors) is a decent start.

Whether we we want to go down the road of changing `Index._data` or `DataFrame._data` can be pursued separately.

**Comment by TomAugspurger** [comment]
> and now unclear what means what.

I think that's @jbrockmendel's reason for changing this in the first place, since `._data` can mean so many things. But with this PR when you need to deal with accessors, or you see `._parent`, you know at a glance what's going on.

**Comment by jreback** [comment]
yeah I think this change is ok. Just want to document this somewhere and quickly survey other uses for comonaility.

**Comment by TomAugspurger** [comment]
Sounds good. http://pandas-docs.github.io/pandas-docs-travis/internals.html#
may be a decent place for documenting these kinds of relationships between
classes and their data attributes.

**Comment by jbrockmendel** [comment]
> Just want to document this somewhere and quickly survey other uses for comonaility.

OK, if we're going down this road, should I update `CategoricalAccessor` to use `_parent` instead of `categories`?  (ATM it is the one case that doesn't use `_data`).

**Comment by jreback** [comment]
yes i am ok with all non internals things changing to use _parent

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
ok, can you add some docs in internals.rst about this

**Pull request merged**